<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8" />
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />




<title>Data Frame</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-5.0.13/css/fa-svg-with-js.css" rel="stylesheet" />
<script src="site_libs/font-awesome-5.0.13/js/fontawesome-all.min.js"></script>
<script src="site_libs/font-awesome-5.0.13/js/fa-v4-shims.min.js"></script>

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>


</head>

<body>

<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
  height: auto;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}

.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>


<div class="container-fluid main-container">

<!-- tabsets -->
<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});
</script>

<!-- code folding -->




<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_').toLowerCase();
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}


.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
  padding-left: 25px;
  text-indent: 0;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>

<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">BRLS-2017</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    <span class="fa fa-gear"></span>
     
    Data Structure
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="intro.html">Introduction</a>
    </li>
    <li>
      <a href="vector.html">Vector</a>
    </li>
    <li>
      <a href="matrix.html">Matrix and Array</a>
    </li>
    <li>
      <a href="dataframe.html">Data Frame</a>
    </li>
    <li>
      <a href="list.html">List</a>
    </li>
    <li>
      <a href="loop.html">Conditions/Loops</a>
    </li>
    <li>
      <a href="function.html">Function</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    <span class="fa fa-gear"></span>
     
    Data Manipulation
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="io.html">Input-Output</a>
    </li>
    <li>
      <a href="apply.html">Apply set of function</a>
    </li>
    <li>
      <a href="dplyr.html">Dplyr</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    <span class="fa fa-gear"></span>
     
    Visualization
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="baseGraphics.html">Base Graphics</a>
    </li>
    <li>
      <a href="ggplot2.html">ggplot2 Graphics</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    <span class="fa fa-gear"></span>
     
    Statistics
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="Descriptive_statistics.html">Descriptive Statistics</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    <span class="fa fa-gear"></span>
     
    Case studies
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="bioconductor.html">Bioconductor</a>
    </li>
    <li>
      <a href="casestudy.html">Case study</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    <span class="fa fa-gear"></span>
     
    Tasks
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="task.html">Tasks</a>
    </li>
    <li>
      <a href="solution.html">Solutions</a>
    </li>
  </ul>
</li>
<li>
  <a href="study.html">StudyMore</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">Data Frame</h1>

</div>


<p>Author: Team BioSakshat</p>
<p>Last update: June 2017</p>
<p>
Copyright © 2017 BioSakshat, Inc. All rights reserved.
</p>
<p>……………………………………………………………..</p>
<div id="gene-expression-data-analysis" class="section level2">
<h2>Gene Expression Data Analysis</h2>
<div id="experimental-setup" class="section level3">
<h3>Experimental setup</h3>
<p>There are two varieties of plant xyz, one is <strong>resistant (R)</strong> to a fungal infection while other is susceptible to the fungal <strong>disease (D)</strong>. The experiment was conducted to study the differential expression of 490 genes (G1, G2, G3,.., G490), which are involved in 3 different pathways (P1, P2 and P3).</p>
<p>The gene expression was measured in the two mentioned varieties of plants, <strong>R</strong> and <strong>D</strong>, on <strong>Day 2, Day 4, Day 6 and Day 8</strong>.</p>
<p>The gene expression values for variety R on these days are given as <strong>R2, R4, R6 and R8</strong> and that for variety D are given as <strong>D2, D4, D6 and D8</strong>.</p>
<p>The gene expression data is stored in a file <a href="data/CaseStudy/path.txt">path.txt</a>. The first 6 rows of the file are shown below.</p>
<div class="figure">
<img src="image/casestudy_fig1.png" />

</div>
</div>
<div id="objective" class="section level3">
<h3>Objective</h3>
<p>Data exploration, visualization and analysis of gene expression data, in a given file, using R to find out those genes, which are differentially expressed on 3 or more days between two varieties, <strong>R</strong> and <strong>D</strong>.</p>
</div>
<div id="steps" class="section level3">
<h3>Steps</h3>
<ol style="list-style-type: decimal">
<li>Import the file “path.txt” into R.</li>
<li>What is the data structure of imported file?</li>
<li>How many rows and columns are there?</li>
<li>What are column names?</li>
<li>Find out minimum, first quantile, median, third quantile, mean and maximum of expression values on each day. Store the result in a file.</li>
<li>Visualization of gene expression on 2,4,6,8 days of D and R plants using boxplot.</li>
</ol>
<div class="figure">
<img src="image/casestudy_fig2.png" />

</div>
<div id="section" class="section level4">
<h4>.</h4>
<ol start="7" style="list-style-type: decimal">
<li>Visualization of pairwise correlation of gene expressions among R2, R4, R6 and R8.</li>
</ol>
<div class="figure">
<img src="image/casestudy_fig3.png" />

</div>
</div>
<div id="section-1" class="section level4">
<h4>.</h4>
<ol start="8" style="list-style-type: decimal">
<li>Visualization of pairwise correlation of gene expressions among D2, D4, D6 and D8.</li>
</ol>
<div class="figure">
<img src="image/casestudy_fig4.png" />

</div>
</div>
<div id="section-2" class="section level4">
<h4>.</h4>
<ol start="9" style="list-style-type: decimal">
<li>Calculate the pairwise correlation coefficient values among R2, R4, R6 and R8.</li>
</ol>
<div class="figure">
<img src="image/casestudy_fig5.png" />

</div>
</div>
<div id="section-3" class="section level4">
<h4>.</h4>
<ol start="10" style="list-style-type: decimal">
<li>Calculate the pairwise correlation coefficient values among D2, D4, D6 and D8.</li>
</ol>
<div class="figure">
<img src="image/casestudy_fig6.png" />

</div>
</div>
<div id="section-4" class="section level4">
<h4>.</h4>
<ol start="11" style="list-style-type: decimal">
<li>Draw a four panel plot depicting four scatterplots of R2 Vs D2, R4 Vs D4, R6 Vs D6 and R8 Vs D8.</li>
</ol>
<div class="figure">
<img src="image/casestudy_fig7.png" />

</div>
</div>
<div id="section-5" class="section level4">
<h4>.</h4>
<ol start="12" style="list-style-type: decimal">
<li>Filter those genes that are up-regulated in D variety on all days i.e. (D2-R2)&gt;0; (D4-R4)&gt;0; (D6-R6)&gt;0 and (D8-R8)&gt;0. Write the differential expression values of these filtered genes in a file, <strong>up.txt</strong>.</li>
</ol>
<p>Ans: 172 genes.</p>
</div>
<div id="section-6" class="section level4">
<h4>.</h4>
<ol start="13" style="list-style-type: decimal">
<li>Count the pathway wise gene count for the genes, which are filtered in step 12.</li>
</ol>
<p>Ans:</p>
<p>P1 P2 P3</p>
<p>106 43 23</p>
</div>
<div id="section-7" class="section level4">
<h4>.</h4>
<ol start="14" style="list-style-type: decimal">
<li>Plot the heatmap showing clustering of genes filtered in step 12. Save the heatmap image.</li>
</ol>
<div class="figure">
<img src="image/casestudy_fig8.png" />

</div>
</div>
<div id="section-8" class="section level4">
<h4>.</h4>
<ol start="15" style="list-style-type: decimal">
<li>You are provided with an annotation file, <a href="data/CaseStudy/anno.txt">anno.txt</a>, of all the genes containing information of gene name, description and accession number. Retrieve the annotations for genes filtered in step 12 from <strong><a href="data/CaseStudy/anno.txt">anno.txt</a></strong> file. Hint: Search “%in%” in help and try to understand from the given example.</li>
</ol>
</div>
<div id="section-9" class="section level4">
<h4>.</h4>
<ol start="16" style="list-style-type: decimal">
<li>Group the genes as per their pathways. Arrange the values for each group according expression on D2. Write the arranged data in a file, <strong>Genes_arranged.txt</strong>.</li>
</ol>
</div>
</div>
</div>
<div id="solution" class="section level2">
<h2>Solution</h2>
<ol style="list-style-type: decimal">
<li>Import the file “path.txt” into R.</li>
</ol>
<pre class="r"><code>data=read.table(&quot;path.txt&quot;,header=T)</code></pre>
<ol start="2" style="list-style-type: decimal">
<li>What is the data structure of imported file?</li>
</ol>
<pre class="r"><code>str(data)</code></pre>
<ol start="3" style="list-style-type: decimal">
<li>How many rows and columns are there?</li>
</ol>
<pre class="r"><code>dim(data)</code></pre>
<ol start="4" style="list-style-type: decimal">
<li>What are column names?</li>
</ol>
<pre class="r"><code>colnames(data)</code></pre>
<ol start="5" style="list-style-type: decimal">
<li>Find out minimum, first quantile, median, third quantile, mean and maximum of expression values on each day. Store the result in a file.</li>
</ol>
<pre class="r"><code>summary(data)</code></pre>
<ol start="6" style="list-style-type: decimal">
<li>Visualization of gene expression on 2,4,6,8 days of D and R plants using boxplot.</li>
</ol>
<pre class="r"><code>boxplot(data[,2:9]);</code></pre>
<ol start="7" style="list-style-type: decimal">
<li>Visualization of pairwise correlation of gene expressions among R2, R4, R6 and R8.</li>
</ol>
<pre class="r"><code>pairs(data[,2:5],upper.panel=NULL)</code></pre>
<ol start="8" style="list-style-type: decimal">
<li>Visualization of pairwise correlation of gene expressions among D2, D4, D6 and D8.</li>
</ol>
<pre class="r"><code>pairs(data[,6:9],upper.panel=NULL)</code></pre>
<ol start="9" style="list-style-type: decimal">
<li>Calculate the pairwise correlation coefficient values among R2, R4, R6 and R8.</li>
</ol>
<pre class="r"><code>rcor=cor(data[,2:5])</code></pre>
<ol start="10" style="list-style-type: decimal">
<li>Calculate the pairwise correlation coefficient values among D2, D4, D6 and D8.</li>
</ol>
<pre class="r"><code>dcor=cor(data[,6:9])</code></pre>
<ol start="11" style="list-style-type: decimal">
<li>Draw a four panel plot depicting four scatterplots of R2 Vs D2, R4 Vs D4, R6 Vs D6 and R8 Vs D8.</li>
</ol>
<pre class="r"><code>par(mfrow=c(2,2))
plot(data$D2,data$R2,xlab=&quot;D2&quot;,ylab=&quot;R2&quot;,cex.lab=1.5);
plot(data$D4,data$R4,xlab=&quot;D4&quot;,ylab=&quot;R4&quot;,cex.lab=1.5);
plot(data$D6,data$R6,xlab=&quot;D6&quot;,ylab=&quot;R6&quot;,cex.lab=1.5);
plot(data$D8,data$R8,xlab=&quot;D8&quot;,ylab=&quot;R8&quot;,cex.lab=1.5);</code></pre>
<ol start="12" style="list-style-type: decimal">
<li>Filter those genes that are up-regulated in D variety on all days i.e. (D2-R2)&gt;0; (D4-R4)&gt;0; (D6-R6)&gt;0 and (D8-R8)&gt;0. Write the differential expression values of these filtered genes in a file, up.txt.</li>
</ol>
<pre class="r"><code>library(dplyr)
# Filter those genes which are up/down regulated in Diseased condition.

temp=mutate(data,diff2=D2-R2,diff4=D4-R4,diff6=D6-R6,diff8=D8-R8,diff2up=diff2&gt;0,diff4up=diff4&gt;0,diff6up=diff6&gt;0,diff8up=diff8&gt;0,totup=diff2up+diff4up+diff6up+diff8up);

head(temp)
up3=filter(temp,totup&gt;3);
write.table(up3,file=&quot;up.txt&quot;,sep=&quot;\t&quot;,eol=&quot;\n&quot;,quote=F,row.names=F)</code></pre>
<ol start="13" style="list-style-type: decimal">
<li>Count the pathway wise gene count for the genes, which are filtered in step 12.</li>
</ol>
<pre class="r"><code>pathcount=table(up3$Path);</code></pre>
<ol start="14" style="list-style-type: decimal">
<li>Plot the heatmap showing clustering of genes filtered in step 12. Save the heatmap image.</li>
</ol>
<pre class="r"><code>library(&quot;gplots&quot;);
h=heatmap.2(as.matrix(up3[,2:9]),trace=&quot;none&quot;)</code></pre>
<ol start="15" style="list-style-type: decimal">
<li>You are provided with an annotation file, <a href="data/CaseStudy/anno.txt">anno.txt</a>, of all the genes containing information of gene name, description and accession number. Retrieve the annotations for genes filtered in step 12 from anno.txt file. Hint: Search “%in%” in help and try to understand from the given example.</li>
</ol>
<pre class="r"><code>annotation=read.table(&quot;anno.txt&quot;, sep=&quot;\t&quot;,header=T)
temp=annotation$Gene  %in% up3$Gene
info=annotation[temp,]</code></pre>
<p>16 Group the genes as per their pathways. Arrange the values for each group according expression on D2. Write the arranged data in a file, Genes_arranged.txt.</p>
<pre class="r"><code>gengrp=group_by(up3, Path)
arr=arrange(gengrp,D2)
write.table(arr,&quot;Genes_arranged.txt&quot;, sep=&quot;\t&quot;, quote = F)</code></pre>
</div>



</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
